src/components/data/RenameTableModal.js

```diff
@@ -31,20 +55,16 @@ const RenameTableModal = ({ onDismiss, onUpdateSuccess, namespace, name, selecte
         withErrorReporting('Error renaming data table.'),
         Utils.withBusyState(setRenaming)
       )(async () => {
-        await Ajax().Metrics.captureEvent(Events.workspaceDataRenameTable, { oldName: selectedDataType, newName })
-        await Ajax().Workspaces.workspace(namespace, name).renameEntityType(selectedDataType, newName)
+        await handleTableRename({ oldName: selectedDataType, newName })
 
-        // Move column settings to new table
-        const oldTableColumnSettingsKey = allSavedColumnSettingsEntityTypeKey({ entityType: selectedDataType })
-        const newTableColumnSettingsKey = allSavedColumnSettingsEntityTypeKey({ entityType: newName })
-        const allColumnSettings = await getAllSavedColumnSettings()
-        const tableColumnSettings = _.get(oldTableColumnSettingsKey, allColumnSettings)
-        if (tableColumnSettings) {
-          await updateAllSavedColumnSettings(_.flow(
-            _.set(newTableColumnSettingsKey, tableColumnSettings),
-            _.unset(oldTableColumnSettingsKey)
-          )(allColumnSettings))
-        }
+        if (renameSetTables) {
+          // The table renames need to happen in sequence. Renaming them in parallel risks running into
+          // a deadlock in the Rawls DB because of row-level locking that occurs during rename.
+          for (const tableName of setTableNames) {
+            await handleTableRename({ oldName: tableName, newName: tableName.replace(selectedDataType, newName) })
+          }
+          await moveTableColumnSettings([{ oldName: selectedDataType, newName }, ..._.map(tableName => ({ oldName: tableName, newName: tableName.replace(selectedDataType, newName) }), setTableNames)])
+        } else await moveTableColumnSettings([{ oldName: selectedDataType, newName }])
```

Comment:
Nit: I think using brackets is more conventional in Terra UI.

```suggestion
        } else {
          await moveTableColumnSettings([{ oldName: selectedDataType, newName }])
        }
```